app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Inertia\Response
+     */
+    public function create(Request $request)
+    {
+        $characters = Auth::user()->characters;
+
+        $code = $request->validate([
+            'code' => "sometimes|string"
+        ])['code']?? null;
+
+        $campaign = Campaign::where('code', $code)->first();
```

Comment:
lets also eager load the users and characters that are already part of the campaign, so the front end will be able to show the characters that have already joined, also we can make the page 404 if they didn't enter the right code with `->firstOrFail()`
```suggestion
        $campaign = Campaign::with('characters', 'users')->where('code', $code)->firstOrFail();
```